docs/0002-atlas-translations-management.rst

```diff
@@ -0,0 +1,136 @@
+
+The ``openedx-app-ios`` uses a clean architecture. Therefore, each app has its own i18n strings files. This is
+useful for Developer Experience, but we believe it's not good for Translator Experience.
+
+Decision
+********
+
+This decision will require few changes to the translation workflow in comparison with the old ``edx-app-ios``:
+
+- Translation files are no longer comitted directly in the mobie app ``openedx-app-ios/android`` repositories
+- Keep the i18n string files split into each clean architecture app without hurting Developer Experience.
+- Identical string IDs across the application will be treated as the same string/translation.
+  For example, ``BUTTON_CLOSE="Close"`` in dashboard is the same ``BUTTON_CLOSE="Close"`` in profile app.
+  Otherwise, a validation script will throw an error and prevent merging the pull request.
+- String files are combined into a single file before preparing it to Transifex
+  (through the `openedx-translations repository`_)
+- Translators can translate the entire app as a single resource
+- Before releasing to App Store, developers need to run a new ``make pull_translations`` command
+- After translations is pulled the translation file is split again into each clean architecture application
+  as if it has been individual resource (a new script)
+
+
+String files combination and splitting process
+==============================================
+This is a new process that's being introduced to have the best combination
+of Developer Experience and Translator Experience.
+
+Translators need to have less resources as possible with strings de-duplicated,
+while Engineers need to have clean architecture with strings split into each app to aid maintainability.
+
+Combining the files during ``combine_translations``
+---------------------------------------------------
+
+However, combining the string files to aid Translator Experience in the following manner:
+
+Suppose we have two apps with one string source file in each::
+
+  Course/Course/en.lproj/Localizable.strings
+  Dashboard/Dashboard/en.lproj/Localizable.strings
+
+This script will collect the strings, remove uneeded comments and combine them in a temporary file::
+
+  I18N/I18N/en.lproj/Localizable.strings
+
+
+This file will be committed into the `openedx-translations repository`_ as described in `OEP-58`_.
+
+This process happens entirely on the CI server (GitHub in this case) after each pull request merge without developer
+intervention.
+
+GitHub Actions CI Script to avoid duplicate string IDs
+------------------------------------------------------
+
+A new script will be created to ensure we don't have duplicate IDs with different content.
+
+For example:
+
+If the Dashboard has ``BUTTON_CLOSE="Close"`` while Course app has ``BUTTON_CLOSE="CLOSE TAB"``, this
+script should fail and require either making the string to match or the ID to be different.
```

Comment:
@volodymyr-chekyrta @brian-smith-tcril any objections against this validation?

For example I see `TITLE` being used multiple times. Any reason it can't be `DASHBOARD.TITLE` and `WHATS_NEW.TITLE`?

https://github.com/openedx/openedx-app-ios/blob/05d3073a822ed12783140aa2744dca939185fe08/Discussion/Discussion/en.lproj/Localizable.strings#L9

https://github.com/openedx/openedx-app-ios/blob/05d3073a822ed12783140aa2744dca939185fe08/WhatsNew/WhatsNew/en.lproj/Localizable.strings#L9

If that's okay, then I believe this part is good to go.

Reply:
Good question.
Duplication of keys in modules is allowed because each module is a separate independent project/service, so it's not a problem for development.
And introducing a new rule for naming could lead to more complexity in the project and more time for onboarding for newcomers, I'd like to avoid it.

Is there any change to store each module's resources separately in their own files without merging them?
It seems that the edX app translations to Transifex are stored in different files with no problem. 
I wonder if we can keep the resources separate.

<img width="1029" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/9251eba5-22c7-4f70-8416-1deec7347a98">

Reply:
> Good question.
> Duplication of keys in modules is allowed because each module is a separate independent project/service, so it's not a problem for development.

Yes, this is very well understood. We will try to not affect clean architecture apps when it comes to developer experience. 

> And introducing a new rule for naming could lead to more complexity in the project and more time for onboarding for newcomers, I'd like to avoid it.

I'll explain what we intend:

 - It's perfectly fine to have duplicate strings that are identical .e.g having `"BUTTON_CLOSE" = "Close"` in multiple apps is fine. 
 - Engineers can focus on their app unaware of the other apps.
 - The only exception is that we'll need to avoid having the same ID used for different strings:

For example the iOS app has only the following strings need fixing:

```
Duplicate IDs with different messages:
ID: "ALERT.ACCEPT"
  Message 1: "Accept";
  Message 2: "ACCEPT";
ID: "TITLE"
  Message 1: "Dashboard";
  Message 2: "Profile";
  Message 3: "Discussions";
  Message 4: "Discover";
  Message 5: "What's New";
ID: "SEARCH.EMPTY_DESCRIPTION"
  Message 1: "Start typing to find the topics";
  Message 2: "Start typing to find the course";
```

While other IDs are completely fine:

```
Duplicate IDs with different messages:
ID: "COURSEWARE.COURSE_CONTENT"
  Message 1: "Course content";
  Message 2: "Course content";
ID: "COURSEWARE.COURSE_UNITS"
  Message 1: "Course units";
  Message 2: "Course units";
ID: "COURSEWARE.NEXT"
  Message 1: "Next";
  Message 2: "Next";
ID: "COURSEWARE.PREVIOUS"
  Message 1: "Prev";
  Message 2: "Prev";
ID: "COURSEWARE.FINISH"
  Message 1: "Finish";
  Message 2: "Finish";
ID: "COURSEWARE.GOOD_WORK"
  Message 1: "Good Work!";
  Message 2: "Good Work!";
ID: "COURSEWARE.BACK_TO_OUTLINE"
  Message 1: "Back to outline";
  Message 2: "Back to outline";
ID: "COURSEWARE.SECTION"
  Message 1: "Section “";
  Message 2: "Section “";
ID: "COURSEWARE.IS_FINISHED"
  Message 1: "“ is finished.";
  Message 2: "“ is finished.";
ID: "COURSEWARE.CONTINUE"
  Message 1: "Continue";
  Message 2: "Continue";
ID: "COURSEWARE.RESUME_WITH"
  Message 1: "Resume with:";
  Message 2: "Resume with:";
ID: "ERROR.RELOAD"
  Message 1: "Reload";
  Message 2: "Reload";
ID: "SEARCH"
  Message 1: "Search";
  Message 2: "Search";
ID: "SEARCH.TITLE"
  Message 1: "Search results";
  Message 2: "Search results";
```

I hope that makes it less of an issue to introduce this lint rule with clear GitHub comments via GitHub Actions it should be an acceptable minimal cost imho.

We can pursue a more complex alternative, but we'd like to start with something simple.

> Is there any change to store each module's resources separately in their own files without merging them?
It seems that the edX app translations to Transifex are stored in different files with no problem.
I wonder if we can keep the resources separate.

Yes, we'd like to change that because we think it will improve Translators Experience.

The new resources will have to be in the [openedx-translations Transifex project](https://app.transifex.com/open-edx/openedx-translations/content/) which currently has 57 resources and the number is growing. 

It's hard to justify adding 10 resources for each app, and require the mobile engineers to follow the [Enable Translations on a New Repository](https://docs.openedx.org/en/latest/developers/how-tos/enable-translations-new-repo.html#id24) every time a new app is created. It's quite of a process and we'd rather save the mobile engineers the effort with some automation.

This proposal ensures that mobile engineers continue with their workflow mostly uninterrupted. 

That was long, I hope it's clearer now.

Reply:
@brian-smith-tcril I'd like to have your input here. The decision here involves combining the resources into a single Transifex resource [similar to the edx-platform](https://github.com/openedx/edx-platform/blob/master/docs/decisions/0018-standarize-django-po-files.rst) to avoid adding too many resources to the transifex project.

It's worth noting that the scope of FC-55 is mostly focused on combining the resources to improve Translators Experience.

Please let me know what do you think a good trade-off is here.

Reply:
@OmarIthawi, I fully understand that this is the only issue among the 9 strings for now. 
It doesn't seem like a big deal to rename them, we can do it, but maybe we can tweak it by adding a `temp_$module_name` prefix to those IDs to handle that via script?
I'm worried about adding a new naming rule because it could make the project more complex. I'd rather keep things simple and accessible for the open-source community.

Reply:
Thanks @volodymyr-chekyrta @brian-smith-tcril I think that's a pretty reasonable way to go.

To confirm, you're suggesting that we only prefix the duplicate strings, and keep the rest unchanged before pushing to Transifex. Right?

I'm asking this because I'm wondering if it makes sense to prefix all strings, and remove the prefix after splitting. 

Both options will remove the need for the lint rule.

Reply:
Hmm 🤔 , maybe adding prefixes to all strings makes sense.
Form something like `$module_name + $string_id`.
@OmarIthawi @brian-smith-tcril what do you think? 
Will it simplify script development?

Reply:
Works for me and as far as I know, it'll be a nice translation experience if someone wants to know the code location of the resource.

I'll update the proposal.